impl/src/main/java/org/ehcache/internal/store/offheap/OffHeapStore.java

```diff
@@ -190,5 +192,39 @@ public void releaseAuthoritativeTier(AuthoritativeTier<?, ?> resource) {
     public void initAuthoritativeTier(AuthoritativeTier<?, ?> resource) {
       initStore(resource);
     }
+
+    @Override
+    public <K, V> LowerCachingTier<K, V> createCachingTier(Configuration<K, V> storeConfig, ServiceConfiguration<?>... serviceConfigs) {
+      return createStore(storeConfig, serviceConfigs);
+    }
+
+    @Override
+    public void releaseCachingTier(LowerCachingTier<?, ?> resource) {
+      if (!createdStores.contains(resource)) {
+        throw new IllegalArgumentException("Given caching tier is not managed by this provider : " + resource);
+      }
+      invalidateAll((OffHeapStore<Object, ?>) resource);
+      releaseStore((Store<?, ?>) resource);
+    }
+
+    private void invalidateAll(OffHeapStore<Object, ?> resource) {
+      OffHeapStore<Object, ?> offheapStore = resource;
+      Set<Object> keys = offheapStore.backingMap().keySet();
+      for (Object key : keys) {
+        try {
+          offheapStore.invalidate(key);
+        } catch (CacheAccessException cae) {
+          // TODO: log? rethrow?
```

Comment:
I actually was wondering what to do as my view clashed with Rishabh's, see https://github.com/ehcache/ehcache3/pull/486#discussion-diff-36517439
Once there's an agreement on what exactly we want to do here I'll make the required changes.